npm_and_yarn/lib/dependabot/npm_and_yarn/file_fetcher.rb

```diff
@@ -11,6 +11,13 @@ module NpmAndYarn
     class FileFetcher < Dependabot::FileFetchers::Base
       require_relative "file_fetcher/path_dependency_builder"
 
+      # Npm always prefixes file paths in the lockfile "version" with "file:"
+      # even when a naked path is used (e.g. "../dep")
+      NPM_PATH_DEPENDENCY_STARTS = %w(file:).freeze
+      # "link:" is only supported by Yarn but is interchangeable with "file:"
+      PATH_DEPENDENCY_STARTS = %w(file: link:. / ./ ../ ~/).freeze
```

Comment:
Do we need the `.` after `link:`?

Reply:
I think so, seems you can register a package name with `yarn link` and then magically reference the linked package name from anywhere with `link:package-name` 😨

Reply:
So not actually a path we can fetch

Reply:
Updating the comment to make this a bit clearer